pvlib/irradiance.py

```diff
@@ -2915,3 +2916,68 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def component_sum_irradiance(solar_zenith,
+                             ghi=None,
+                             dhi=None,
+                             dni=None,
+                             dni_clear=None):
+    r"""
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+
+    .. math::
+
+       GHI = DHI + DNI \cos(\theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
```

Comment:
I just kept it the variable as solar_zenith, and made the description more general so it can be either solar_zenith or apparent_zenith. That should help resolve the apparent_zenith/solar_zenith conversation here @cwhanse and @kanderso-nrel